builtin/reset.c

```diff
@@ -454,9 +454,11 @@ int cmd_reset(int argc, const char **argv, const char *prefix)
 			int flags = quiet ? REFRESH_QUIET : REFRESH_IN_PORCELAIN;
 			if (read_from_tree(&pathspec, &oid, intent_to_add))
 				return 1;
-			if (get_git_work_tree())
+			if (get_git_work_tree()) {
 				refresh_index(&the_index, flags, NULL, NULL,
 					      _("Unstaged changes after reset:"));
+				the_index.reset_mixed = 1;
```

Comment:
Is this going to start colliding with Duy's work to remove `the_index` everywhere? Is there something we can do now to prepare for that coming from upstream?

Reply:
I'm sure it will but he would need to change the `refresh_index` method above as well so at that point we can just create the fixup commit to fix our use of `the_index` as well.